spirv_hlsl.cpp

```diff
@@ -3041,6 +3127,43 @@ void CompilerHLSL::emit_function_prototype(SPIRFunction &func, const Bitset &ret
 			var->parameter = &arg;
 	}
 
+	if (func.self == ir.default_entry_point && get_entry_point().model == ExecutionModelGeometry)
+	{
+		uint32_t input_vertices = 1;
+		auto &execution = get_entry_point();
+		if (execution.flags.get(ExecutionModeInputLines))
+			input_vertices = 2;
+		else if (execution.flags.get(ExecutionModeInputLinesAdjacency))
+			input_vertices = 4;
+		else if (execution.flags.get(ExecutionModeInputTrianglesAdjacency))
+			input_vertices = 6;
+		else if (execution.flags.get(ExecutionModeTriangles))
+			input_vertices = 3;
+
+		string prim;
```

Comment:
This doesn't need to be a string, const char * is good.

Reply:
Done